template/{% if git_platform=="github.com" %}.github{% endif %}/workflows/ci.yml.jinja

```diff
@@ -27,29 +27,36 @@ jobs:
       python-version: ${{ matrix.python-version }}
     secrets:
       CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
-{% endraw %}{% if docker %}
+{% endraw %}{% if docker %}{% raw %}
   container:
+    needs: test
+    if: always()
     uses: ./.github/workflows/_container.yml
+    with:
+      publish: ${{ needs.test.result == 'success' }}
     permissions:
       contents: read
       packages: write
-{% endif %}{% if sphinx %}
+{% endraw %}{% endif %}{% if sphinx %}{% raw %}
   docs:
     uses: ./.github/workflows/_docs.yml
-{% endif %}
+    with:
+      publish: ${{ needs.test.result == 'success' }}
+
+{% endraw %}{% endif %}
```

Comment:
Ermm, the docs publish is dependent on a `needs.test.result` but `test` is no longer in `needs`. What does that do?